### PR TITLE
DocVision 모델 추가

### PR DESCRIPTION
**Pull request by hancheolcho**
# 내용

DocVision 모델을 추가하였습니다.

lmms-eval는 Model Wrapper (`lmms_eval/models/docvision.py`)를 구현할 때, generate_until 메소드 (답변 생성형 태스크용)와 loglikelihood 메소드 (logits 이용한 분류형 태스크)를 구현하도록 되어있는데요.
현재는 generate_until 메소드만 구현해 놓은 상태입니다.
따라서 docvqa, infovqa, textvqa, chartqa, mpdocvqa 등의 태스크는 현재 평가용으로 사용 가능하며, mmlu 같은 태스크는 평가가 안됩니다.

평가를 위한 conda 환경 셋팅 및 평가 스크립트 사용법은 README.md에 추가했습니다.

DocVision 모델 구현체는 `lmms_eval/models/docvision_utils` 디렉토리 아래에 있으며, import 구문을 현재 위치에 맞게 변경하고, sequence parallel 부분 및 distributed.barrier() 호출 부분을 삭제한 부분이 있습니다.

image
image

### CODE REVIEW

**Review by seunggu-upstage — COMMENTED** [review]
엄청난 trial and error가 느껴지네요...몇 가지 코멘트 남겼습니다!

File: lmms_eval/models/docvision.py
[the remark this answers is not in the document]
Comment (hancheolcho) [reply lmms_eval/models/docvision.py:1]: > L77에서 _eos_token_id를 얻을 때 None을 tokenizer.Encode 해서 얻는 것 같은데 어떤 의미일까요?

lmms-eval 에서 모델을 구현할 때에 다양한 정보들을 class property 형태로 구현을 했더라구요.
generate_until() 메소드에서 gen_kwargs에 만들 때에 해당 토큰을 주도록 되어 있었습니다.
image


> L297부터 주석처리 된 부분은 어떤 역할이었을까요? 뭔가 cont가 except에서 만들어진 것이다 보니 error handling 관련일 것 같은데, 불필요하다면 지워도 좋을 것 같습니다!

이건 InstructBLIP에 있던 부분이라 지워도 될 것 같네요.
삭제하겠습니다.

File: lmms_eval/models/docvision_utils/pipeline_initialize.py
[the remark this answers is not in the document]
Comment (hancheolcho) [reply lmms_eval/models/docvision_utils/pipeline_initialize.py:1]: 모델 관련 코드 수정은 최소한으로 하려고 했습니다.
나중에 코드가 업데이트 될 경우에 다시 작업해야하기도 해서요.

torch.distributed.barrier() 명령이 빠진 부분들은, 해당 명령이 있으면 모델 실행이 안되는 부분들입니다.
우리 모델의 경우에 deepspeed를 직접 사용했었지만 lmms-eval 의 경우에는 HF Accelerate를 사용하기 때문에, 호환이 안되는 부분들로 보입니다.

map_location="cpu"를 포함해서 Accelerate가 자동으로 처리하는 항목들은 제거를 했습니다.
이건 cpu로 줘도 될 것 같긴 합니다.
현재 gpu가 없어서 테스트가 안되니, issue에 등록해놓겠습니다.

File: lmms_eval/models/docvision_utils/pipeline_utils.py
[the remark this answers is not in the document]
Comment (hancheolcho) [reply lmms_eval/models/docvision_utils/pipeline_utils.py:1]: L350-372에서 MoRA adapter를 적용하고 있습니다.

**Review by seunggu-upstage — APPROVED** [review]
답변 감사합니다! LGTM :-)
